functions/src/test/mocks.ts

```diff
@@ -213,6 +213,14 @@ export const mockUsers = {
     }
 }
 
+export const storageBucketMock = {
+    file: (path: String) => ({
+         // We can pass 'not_exist' as the UID to test if a file exists:
+        exists: () => Promise.resolve(path.split('/').indexOf('not_exists') === -1),
+        delete: () => { exports.spyOnMe2(); Promise.resolve(true) }
```

Comment:
Should this be returning the promise?

Reply:
Yes - exists in the actual api returns a promise which resolves to true/false. In this case you can call it on the file reference without getting the file

Reply:
To be specific in case there's confusion, though I'm sure you know this :P
```js
const foo = () => Promise.resolve('bar')
foo() // Promise
foo().then(console.log) // 'bar'
```